include/aie/Dialect/AIE/IR/AIE.td

```diff
@@ -75,23 +75,44 @@ def AnyScalar : TypeConstraint<Or<[Index.predicate,
                                   "scalar">;
 
 
-def AIE_TargetOp: AIE_Op<"target", []>, Results<(outs Index)> {
-  let summary = "Define AIE target such as architecture";
+def AIE_DeviceOp: AIE_Op<"device", [AIEArray,
+                                    HasParent<"ModuleOp">,
+                                    SingleBlockImplicitTerminator<"EndOp">]> {
+  let summary = "Define an AIE design targetting a complete device";
   let description = [{
-    This operation allows specifications of the target such as the AIE architecture. Currently,
-    we support targets for AIE architecture for AIE and AIE2.
+    This operation describes a design that executes on a particular AIEngine device.
+    It exists at the toplevel of a design; although currently it does not replace the
+    default toplevel module in MLIR, the intention is that this could be the case
+    in the future.
+
+    When using this operation, all resources in a physical device are available and
+    the design does not need to be concerned with other potential users of a physical
+    device.  In addition, within an "aie.device" operation, tile addresses are absolute
+    coordinates and are not intended to describe a relocatable design.  To describe
+    a portion of a device which may be relocatable, see the "aie.segment" operation.
+    The design itself is described using a region of code contained by the device
+    operation.
```

Comment:
I don't think `aie.segment` is added by this PR. Would it will also implement `AIEArray`? Or maybe the last sentence suggests a different design?

Reply:
Yeah, that's basically the idea.